File: mac/Classes/Controllers/MainWindow.m

#import "MainWindow.h"


@implementation MainWindow

- (BOOL)makeFirstResponder:(NSResponder *)aResponder {
    BOOL result = [super makeFirstResponder:aResponder];
    if (result) {
        id delegate = [self delegate];
        if ([delegate respondsToSelector:@selector(window:didChangeFirstResponder:)]) {
            [delegate window:self didChangeFirstResponder:aResponder];
        }
    }
    return result;
}

@end

File: LiveReload/Source/Presentation/MainWindow.swift
import Cocoa

public class MainWindow: NSWindow {

    public static let didChangeFirstResponder = "\(MainWindow.self).didChangeFirstResponder"

    public override func makeFirstResponder(aResponder: NSResponder?) -> Bool {
        let result = super.makeFirstResponder(aResponder)
        if result {
            NSNotificationCenter.defaultCenter()
                .postNotificationName(MainWindow.didChangeFirstResponder, object: self)
        }
        return result
    }

}
